Telemetry/Telemetry.swift

```diff
@@ -81,6 +90,35 @@ open class Telemetry {
             log.debug("Ping response: \(response.response?.statusCode ?? -1).")
         }
     }
+
+    private static func commonPingFormat(forType type: TelemetryDocType) -> [String: Any] {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "yyyy-MM-dd"
+        let date = formatter.string(from: NSDate() as Date)
+        let displayVersion =  [
+            AppInfo.appVersion,
+            "b",
+            AppInfo.buildNumber
+        ].joined()
+        let version = ProcessInfo.processInfo.operatingSystemVersion
+        let osVersion = "\(version.majorVersion).\(version.minorVersion).\(version.patchVersion)"
+
+        return [
+            "type": type.rawValue,
+            "id": NSUUID().uuidString,
```

Comment:
NSUUID inherits from NSObject so not entirely sure what NSUUID() would be here. Just to be safe I use the string method.